llvm/lib/CheerpWriter/CheerpWasmWriter.cpp

```diff
@@ -649,6 +649,25 @@ void CheerpWasmWriter::encodeExtractLane(WasmBuffer& code, const llvm::ExtractEl
 		encodeInst(WasmSIMDU32Opcode::F32x4_EXTRACT_LANE, index, code);
 	else if (elementType->isDoubleTy())
 		encodeInst(WasmSIMDU32Opcode::F64x2_EXTRACT_LANE, index, code);
+	else if (elementType->isIntegerTy(1))
+	{
+		// For boolean vectors, we have to know how many elements the vector has.
+		const FixedVectorType* vecType = cast<FixedVectorType>(eei.getVectorOperand()->getType());
+		const unsigned num = vecType->getNumElements();
+		if (num == 2)
+		{
+			encodeInst(WasmSIMDU32Opcode::I64x2_EXTRACT_LANE, index, code);
+			encodeInst(WasmOpcode::I32_WRAP_I64, code);
```

Comment:
Potentially here you could also consider just extracting i32x4_EXTRACT_LANE(index*2) (should be equivalent).

Reply:
Done, nicely spotted!